code/game/machinery/outpost_electrolyzer.dm

```diff
@@ -0,0 +1,540 @@
+//allows production of hydrogen from ice chunks
+#define MOLS_PER_ICE 50 //1 ice = 50 mols
+#define MOLS_PER_MERIT 10 //10 mols = 1 merit
+#define MERITS_PER_ICE MOLS_PER_ICE / MOLS_PER_MERIT //1 ice = 5 merits
+#define MERITS_USED_PER_TICK 2
+#define H2_PUMP_SHUTOFF_PRESSURE 4000
+#define CREDITS_TO_MERITS 4 // currently 2:5 credits to mols hydrogen. # of credits per merit
+#define OUTPOST_HYDROGEN_CUT 0.8
+#define HYDROGEN_IDEAL 45000 //used for high and low end of merit multiplier
+#define MERIT_EXPONENT 0.95 //used for diminishing returns, values closer to 1 increase returns, lower decrease.
+
+/obj/machinery/mineral/electrolyzer_unloader
+	name = "ice unloading machine"
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "unloader"
+	density = TRUE
+	input_dir = WEST
+	output_dir = EAST
+	needs_item_input = TRUE
+	processing_flags = START_PROCESSING_MANUALLY
+
+/obj/machinery/mineral/electrolyzer_unloader/pickup_item(datum/source, atom/movable/target, atom/oldLoc)
+	if(istype(target, /obj/structure/ore_box))
+		var/obj/structure/ore_box/box = target
+		for(var/obj/item/stack/ore/ice/chunk in box)
+			unload_mineral(chunk)
+	else if(istype(target, /obj/item/stack/ore/ice))
+		var/obj/item/stack/ore/chunk = target
+		unload_mineral(chunk)
+
+// electrolyzer + console
+
+/obj/machinery/computer/electrolyzer_console
+	name = "electrolyzer console"
+	desc = "Deposits hydrogen merits, with 20% going to outpost upkeep."
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "console"
+
+	var/obj/machinery/mineral/electrolyzer/linked_electrolyzer
+
+/obj/machinery/computer/electrolyzer_console/Initialize()
+	. = ..()
+	find_electrolyzer()
```

Comment:
this means that any of these spawned or built after roundstart will automatically try to link, if you don't want that you can apply this change, if you're fine with it just resolve this
```suggestion
/obj/machinery/computer/electrolyzer_console/Initialize(mapload)
	. = ..()
	if(mapload)
		find_electrolyzer()
```